examples/basic.rs

```diff
@@ -68,32 +65,12 @@ async fn main() -> Result<()> {
         _c: String,
     }
 
-    if let Some(rows) = session
-        .query("SELECT a, b, c FROM examples_ks.basic", &[])
-        .await?
-        .rows
-    {
-        for row_data in rows.into_typed::<RowData>() {
-            let row_data = row_data?;
-            println!("row_data: {:?}", row_data);
-        }
-    }
-
-    // Or simply as untyped rows
-    if let Some(rows) = session
+    let result = session
         .query("SELECT a, b, c FROM examples_ks.basic", &[])
-        .await?
-        .rows
-    {
-        for row in rows {
-            let a = row.columns[0].as_ref().unwrap().as_int().unwrap();
-            let b = row.columns[1].as_ref().unwrap().as_int().unwrap();
-            let c = row.columns[2].as_ref().unwrap().as_text().unwrap();
-            println!("a, b, c: {}, {}, {}", a, b, c);
-
-            // Alternatively each row can be parsed individually
-            // let (a2, b2, c2) = row.into_typed::<(i32, i32, String)>() ?;
-        }
+        .await?;
+    let mut iter = result.rows_typed::<(i32, i32, String)>()?;
+    while let Some(row_data) = iter.next().transpose()? {
+        println!("row_data: {:?}", row_data);
     }
 
     let metrics = session.get_metrics();
```

Comment:
1. The example deleted intentionally was one with untyped rows. It should be `rows_typed::<RowData>`, this was indeed an error.
2. The thing is, deserialization refactor is all about moving to iterators to avoid allocations. `row` and `rows` are inherently allocated collections, that's why we discourage from using them at this point and remove altogether at further point, when new deserialization API is introduced.

Reply:
Will there by no way to operate on untyped structs? So no methods like `as_int()`? If so, I suspect we will encounter similar problems as with serialization refactor - our users have use cases that we didn't expect that require those dynamic capabilities.

Reply:
At least in the current state of the refactor, the legacy deserialization framework is still accessible, as a layer (mainly consisting of allocations) over the new framework.
If some people with problematic use cases pop up, we might decide to keep the legacy layer for a long time (or even forever...). It's still wise not to expose those discouraged dynamic capabilities in the doc book; or, if we really have to, rather in the migration guide only.

Reply:
For us not to forget about this possible dynamic usecase that need to be taken care of in the further stages of the refactor, I've restored the mentioned example.